imgui_demo.cpp

```diff
@@ -6030,22 +6030,44 @@ void ImGui::ShowStyleEditor(ImGuiStyle* ref)
             if (style.CurveTessellationTol < 0.10f) style.CurveTessellationTol = 0.10f;
 
             // When editing the "Circle Segment Max Error" value, draw a preview of its effect on auto-tessellated circles.
-            ImGui::DragFloat("Circle Segment Max Error", &style.CircleSegmentMaxError, 0.01f, 0.10f, 10.0f, "%.2f");
+            ImGui::DragFloat("Circle Tessellation Max Error", &style.CircleTessellationMaxError , 0.005f, 0.10f, 10.0f, "%.2f", ImGuiSliderFlags_AlwaysClamp);
             if (ImGui::IsItemActive())
             {
                 ImGui::SetNextWindowPos(ImGui::GetCursorScreenPos());
                 ImGui::BeginTooltip();
-                ImVec2 p = ImGui::GetCursorScreenPos();
+                ImGui::TextUnformatted("N - number of segments");
+                ImGui::TextUnformatted("R - radius");
+                ImGui::Spacing();
                 ImDrawList* draw_list = ImGui::GetWindowDrawList();
-                float RAD_MIN = 10.0f, RAD_MAX = 80.0f;
-                float off_x = 10.0f;
-                for (int n = 0; n < 7; n++)
+                const float min_widget_width = ImGui::CalcTextSize("N: MM\nR: MM.MM").x;
+                float RAD_MIN = 5.0f, RAD_MAX = 80.0f;
+                for (int n = 0; n < 9; n++)
                 {
-                    const float rad = RAD_MIN + (RAD_MAX - RAD_MIN) * (float)n / (7.0f - 1.0f);
-                    draw_list->AddCircle(ImVec2(p.x + off_x + rad, p.y + RAD_MAX), rad, ImGui::GetColorU32(ImGuiCol_Text), 0);
-                    off_x += 10.0f + rad * 2.0f;
+                    const float rad = RAD_MIN + (RAD_MAX - RAD_MIN) * (float)n / (9.0f - 1.0f);
+
+                    const int segment_count = draw_list->_CalcCircleAutoSegmentCount(rad);
```

Comment:
imgui_demo never relies on private APIs, except this one new case.

https://github.com/ocornut/imgui/blob/00d570e280533bca9cf6e5652d0f2eb131d032e7/imgui.h#L2440-L2447

When porting over the demo, I had to skip this line:

https://github.com/oprypin/crystal-imgui/commit/7381050dec5949e5e7bc4eccab82f8af5b31699c#diff-03b057dd391cd708d0edc96be17b8790d6471aac4adf0ec433745795795a99d3R5588

I think this should be fixed

Reply:
It is documented as private but available in imgui.h and we use it to provide info about tessellation in the demo. I think that's fine, and very obviously we don't expect people to copy/reuse this overly specific tessellation display code.

Reply:
Please consider that previously it would be 100% true to say "absolutely everything in imgui_demo can be done without internal calls". I think that's worth something. But now you cannot say this.